STYLE_GUIDE.md

```diff
@@ -0,0 +1,1245 @@
+  * [DO use as few HRX files as possible for a given feature](#do-use-as-few-hrx-files-as-possible-for-a-given-feature)
+  * [DON'T have HRX files longer than 500 lines or so](#dont-have-hrx-files-longer-than-500-lines-or-so)
+  * [DO use descriptive paths](#do-use-descriptive-paths)
+  * [DO put error specs in a separate "error" directory](#do-put-error-specs-in-a-separate-error-directory)
+  * [PREFER underscore-separated paths](#prefer-underscore-separated-paths)
+* [Testing](#testing)
+  * [DO test only one thing per spec](#do-test-only-one-thing-per-spec)
+  * [DO use type selectors as placeholders](#do-use-type-selectors-as-placeholders)
+  * [DO use descriptive names for multiple placeholders](#do-use-descriptive-names-for-multiple-placeholders)
+  * [DO use single-letter names for irrelevant placeholders](#do-use-single-letter-names-for-irrelevant-placeholders)
+  * [DO use style rules for expression-level tests](#do-use-style-rules-for-expression-level-tests)
+  * [PREFER making imported files partials](#prefer-making-imported-files-partials)
+  * [DO name single imported or used files "other"](#do-name-single-imported-or-used-files-other)
+  * [PREFER referring to issues when marking specs as `:todo`](#prefer-referring-to-issues-when-marking-specs-as-todo)
+* [Documentation](#documentation)
+  * [CONSIDER adding a comment explaining your spec](#consider-adding-a-comment-explaining-your-spec)
+  * [CONSIDER adding README.md files to parent directories](#consider-adding-readme.md-files-to-parent-directories)
+  * [DON'T write HRX comments](#dont-write-hrx-comments)
+* [Formatting](#formatting)
+  * [DO use comment dividers to separate specs within an HRX file](#do-use-comment-dividers-to-separate-specs-within-an-hrx-file)
+  * [DO order files consistently](#do-order-files-consistently)
+  * [DO use Dart Sass's output as the default](#do-use-dart-sasss-output-as-the-default)
+  * [DO end each file with a newline](#do-end-each-file-with-a-newline)
+  * [DO use two spaces for indentation](#do-use-two-spaces-for-indentation)
+  * [PREFER one-line rules when they contain a single child with no children](#prefer-one-line-rules-when-they-contain-a-single-child-with-no-children)
+  * [DON'T use one-line rules when they contain more than one child](#dont-use-one-line-rules-when-they-contain-more-than-one-child)
+  * [DON'T use one-line rules when they contain grandchildren](#dont-use-one-line-rules-when-they-contain-grandchildren)
+
+## Organization
+
+### DO write specs in HRX files
+
+<details>
+<summary>Example</summary>
+
+#### Good
+
+`nesting.hrx`
+
+```hrx
+<===> single/input.scss
+a {
+  b {c: d}
+}
+
+<===> single/output.css
+a b {
+  c: d;
+}
+```
+
+#### Bad
+
+`nesting/single/input.scss`
+
+```scss
+a {
+  b {c: d}
+}
+```
+
+`nesting/single/output.css`
+
+```css
+a b {
+  c: d;
+}
+```
+
+</details>
+
+All specs should be written in HRX files unless they contain invalid UTF-8
```

Comment:
Maybe mention that the filenames don't need to be UTF-8, since that was brought up in #1365

Reply:
The filename in that issue *is* UTF-8. I believe [Git itself expects paths to be UTF-8 encoded](https://git-scm.com/docs/git-commit#_discussion), so I don't think we need to be explicit about how they're handled here.